### PR TITLE
ENH: Faster optimization loop in region based dealiasing + extra continuity term

### PR DESCRIPTION
**Pull request by rcjackson**
I have reimplemented the optimization loop in the region based dealiasing to be faster and to have an extra term that adds a cost for correcting a region too far from the one with the closest mean velocity (i.e. it keeps regions of similar velocity together to better handle cases that may deviate far from the sounding).

### CODE REVIEW

**Comment by vlouf** [comment]
@scollis No I don't.

**Comment by scollis** [comment]
Looks like usual problem @rcjackson .. 
When setting tolerances think about physical reasonableness. precision varies by machine

**Comment by rcjackson** [comment]
Nah, I had enforced the continuity term a bit too strong...when I tested it I ran test_dealias.py instead of test_region_dealias.py. Easy to get those two confused!

**Comment by rcjackson** [comment]
I mainly did this to make this code more usable for dealiasing the XSAPR sector scans. My goal is to hopefully have an example of those in my PyDDA talk for AMS (like I had in the abstract), but if not I have other stuff too.

**Comment by scollis** [comment]
Testing ftw 

-sent from a mobile device-

**Comment by scollis** [comment]
@zssherman can you check this works

**Comment by zssherman** [comment]
@scollis Sounds good, I will check it.

File: pyart/correct/region_dealias.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def dealias_region_based(
     # perform dealiasing
     vdata = radar.fields[vel_field]['data'].view(np.ndarray)
     data = vdata.copy()     # dealiased velocities
-
+    mean_vel_last = np.nan
```
Comment (zssherman) [inline pyart/correct/region_dealias.py]: From what I'm seeing, this isn't used anywhere.

File: pyart/correct/region_dealias.py [hunk 2]
```diff
@@ -445,20 +467,30 @@ def _cost_function(nyq_vector, regions, vels_slice,
...
+
+        # Regional continuity
+        vels_without_cur = np.delete(vels_slice_means, reg)
+        diffs = np.square(vels_slice_means[reg]-vels_without_cur)
+        the_min = np.argmin(diffs)
+        add_value2 = (vels_slice_means[reg] +
+                      np.round(nyq_vector[i])*v_nyq_vel -
+                      vels_without_cur[the_min])
+
+        if(np.isfinite(add_value2)):
```
Comment (zssherman) [inline pyart/correct/region_dealias.py]: I believe there is 3 of these extra parenthesis throughout the code, I would just make it:
if np.isfinite(add_values2)

**Comment by scollis** [comment]
Restarting AppVeyor

**Comment by scollis** [comment]
Thanks all! Merging

**Pull request merged**